pycbc/inference/models/tools.py

```diff
@@ -728,11 +728,14 @@ def reconstruct(self, rec=None, seed=None):
         if rec is None:
             rec = {}
 
-        def get_loglr():
-            p = self.current_params.copy()
-            p.update(rec)
-            self.update(**p)
-            return self.loglr
+        if set_loglr is None:
```

Comment:
@WuShichao The flags used here interact with the output of the loglr function. You should make sure you undersand how that works to determine how the function you pass in should behave. Will it behave correctly in these cases? You know the primary model on its own will, but you seem to pass your models loglr function. Does it respect these flags properly? I'm not sure your solution has through through this yet.

Reply:
Now I pass a function to it.